libs/libc/uuid/lib_uuid_from_string.c

```diff
@@ -0,0 +1,113 @@
+/****************************************************************************
+ * libs/libc/uuid/lib_uuid_from_string.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <string.h>
+#include <uuid.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * uuid_from_string
+ *
+ * Description:
+ *   convert a string representation of an UUID into a binary representation.
+ *
+ * See also:
+ *   http://www.opengroup.org/onlinepubs/009629399/uuid_from_string.htm
+ *
+ ****************************************************************************/
+
+void uuid_from_string(const char *s, uuid_t *u, uint32_t *status)
+{
+  int n;
+
+  /* Short-circuit 2 special cases: NULL pointer and empty string. */
+
+  if (s == NULL || *s == '\0')
+    {
+      uuid_create_nil(u, status);
+      return;
+    }
+
+  /* Assume the worst. */
+
+  if (status != NULL)
+    {
+      *status = uuid_s_invalid_string_uuid;
+    }
+
+  /* The UUID string representation has a fixed length. */
+
+  if (strlen(s) != UUID_STR_LEN)
```

Comment:
Should we protect this with `strnlen(s, UUID_STR_LEN+1)`

Reply:
Done.